step-strikers-app/step-strikers-app/GeneralUIViewControllerExtensions.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  GeneralUIViewControllerExtensions.swift
+//  step-strikers-app
+//
+//  Created by Nicholas Huang on 2/28/23.
+//
+
+import UIKit
+import FirebaseFirestore
+
+extension UIViewController: UITextFieldDelegate {
+    
+    func assignBackground() {
+        let background = UIImage(named: "Background")
+        var imageView: UIImageView!
+        imageView = UIImageView(frame: self.view.frame)
+        imageView.clipsToBounds = true
+        imageView.image = background
+        imageView.center = view.center
+        view.addSubview(imageView)
+        view.sendSubviewToBack(imageView)
+    }
+    
+    func createLabel(x:Int, y:Int, w:Int, h:Int, font:String, size:CGFloat, text:String) {
+        let label = UILabel(frame: CGRect(x:x, y:y, width:w, height:h))
+        label.textAlignment = .right
+        label.text = text
+        label.font = UIFont(name:font, size:size)
+        view.addSubview(label)
+    }
+    
+    func createTextField(x:Int, y:Int, w:Int, h:Int, secured:Bool) -> UITextField! {
+        let field = UITextField(frame: CGRect(x:x, y:y, width:w, height:h))
+        field.backgroundColor = UIColor.white
+        field.borderStyle = UITextField.BorderStyle.roundedRect
+        field.isSecureTextEntry = secured
+        field.autocapitalizationType = .none
+        field.delegate = self
```

Comment:
I really don't understand why there is a delegate here? I don't see why you would need to delegate it.